File: src/dataset/cpp/difference-of-squares_exercism-tqa236/gemini/GeminiFlash_difference_of_squares_v4.cpp
#include "difference_of_squares.h"

int difference_of_squares::square_of_sum(int num) {
  long long sum = static_cast<long long>(num) * (num + 1) / 2;
  return sum * sum;
}

int difference_of_squares::sum_of_squares(int num) {
  return num * (num + 1) * (2 * num + 1) / 6;
}

int difference_of_squares::difference(int num) {
  return square_of_sum(num) - sum_of_squares(num);
}

File: src/dataset/python/difference-of-squares_exercism-ThomasZumsteg/gemini/GeminiFlash_difference_of_squares_v4.py
"""Function for calculating squares of sums or sums
of squares and differences between the two
"""
def square_of_sum(num):
    """Square of sum of numbers less then or equal to [num]"""
    n = num + 1
    return (num * n // 2) ** 2

def sum_of_squares(num):
    """Sum of square of numbers less then or equal [num]"""
    return num * (num + 1) * (2 * num + 1) // 6

def difference(num):
    """Difference between the square_of_sums and sum_of_square"""
    n = num + 1
    return num * (n * (num * n - 1) - (2 * num + 1)) // 6